arch/xtensa/core/cache.c

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/arch/cpu.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/toolchain.h>
+
+#include <zephyr/arch/xtensa/cache.h>
+#include <zephyr/cache.h>
+
+#ifdef CONFIG_DCACHE
+
+void arch_dcache_enable(void)
+{
+	/* not implemented */
+}
+
+int arch_dcache_range(void *addr, size_t size, int op)
+{
+	if (op == K_CACHE_INVD) {
+		z_xtensa_cache_inv(addr, size);
+	} else if (op == K_CACHE_WB) {
+		z_xtensa_cache_flush(addr, size);
+	} else if (op == K_CACHE_WB_INVD) {
+		z_xtensa_cache_flush_inv(addr, size);
+	} else {
+		return -ENOTSUP;
+	}
+	return 0;
```

Comment:
useless `return 0` ?

Reply:
/home/nashif/zephyrproject/zephyr/arch/xtensa/core/cache.c:34:1: warning: control reaches end of non-void function [-Wreturn-type]
   34 | }
      | ^

Reply:
Right, nvm.

Reply:
it certainly isn't useless - every `if` branch before the last `else` proceeds to below the whole`if`-`else` so it needs a `return`